### ISSUE TITLE
Update to newer version of pydantic required

### ISSUE THREAD

**Issue by cqc-melf** [Qiskit/qiskit-ibm-runtime#1514]
**Describe the bug**
When installing qiskit_ibm_runtim == 0.21.0 it looks like it is compatible with `pydantic==1.10.13` but when trying to run something it is fails with:
ImportError: cannot import name 'ValidationInfo' from 'pydantic' 
Might be good to update to a newer version.

**Steps to reproduce**
```
pip install pydantic==1.10.13
pip install qiskit_ibm_runtime==0.21.0
from qiskit_ibm_runtime import QiskitRuntimeService
```

**Suggested solutions**
update the version requirement of pydantic

- **qiskit-ibm-runtime version**: 0.21.0
- **Python version**: 3.11
- **Operating system**: linux

**Comment by kt474** [comment]
Thanks for reporting this! This was fixed in #1500, `pydantic` was missing from our `requirements.txt`

**Comment by cqc-melf** [comment]
Great! Thank you!
